config/config.go

```diff
@@ -911,7 +911,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.gumgum.endpoint", "https://g2.gumgum.com/providers/prbds2s/bid")
 	v.SetDefault("adapters.improvedigital.endpoint", "http://ad.360yield.com/pbs")
 	v.SetDefault("adapters.inmobi.endpoint", "https://api.w.inmobi.com/showad/openrtb/bidder/prebid")
-	v.SetDefault("adapters.interactiveoffers.endpoint", "https://prebid-server.ioadx.com/bidRequest/?partnerId=d9e56d418c4825d466ee96c7a31bf1da6b62fa04")
+	v.SetDefault("adapters.interactiveoffers.endpoint", "http://localhost:8020/bidRequest/?partnerId={{.AccountID}}")
```

Comment:
Please do not forget to modify it here too

Reply:
And localhost :)

Reply:
I keep getting this error

F0817 08:41:39.043591   11199 main.go:37] Configuration could not be loaded or did not pass validation: validation errors (1 error):
  1: Unable to resolve endpoint: https://prebid-server.ioadx.com/bidRequest/?partnerId={{.PartnerId}} for adapter: interactiveoffers. template: endpointTemplate:1:56: executing "endpointTemplate" at <.PartnerId>: can't evaluate field PartnerId in type macros.EndpointTemplateParams


And checking the file macros/macros.go 

// EndpointTemplateParams specifies params for an endpoint template
type EndpointTemplateParams struct {
	Host        string
	PublisherID string
	ZoneID      string
	SourceId    string
	AccountID   string
	AdUnit      string
}

I'm missing something...

Reply:
I believe "Please do not forget to modify AccountId to PartnerId" is referring to `interactiveoffers.go` line 116, which is currently correct. In the URL template above it should stay `AccountID`, as the macro function doesn't allow for `PartnerId` as a parameter.  It is a little confusing, as the macros package doesn't allow for arbitrary endpoint macro substitutions, and `AccountId` already has another meaning in the code. The reason we don't allow arbitrary macros is so at run time PBS can validate that the URL templates supplied resolve into valid URLs, which requires us to supply dummy values for all the parameters. This can only be done with a known set of macros. So as long as your `PartnerId` resembles an `AccountId` then reusing `AccountId` will work. 

A better way of accomplishing your goal would be to use the html library functions to add query string parameters to the URL instead of trying to resolve it as a macro. Or even a simple string substitution. An empty/missing query string will still resolve as a valid URL (it checks that the URL is parseable, not that it will actually connect). The provided template routines are only needed for parameters that appear in the domain/path part of the URL, so we can properly verify that the URL is a valid URL.

Reply:
Thank you for the answer @hhhjort! 
@IOTiagoFaria I apologize for my previous comment, I didn't check if it works all together. You were right about `AccountId`

Reply:
Thank you both for the explanation, if its working the way it is and theres no code issues I'll keep it as it is.

I've also changed the java version and on that I've made a simple string replace and the code didn't pass some checks, I had to use some kind of macro...

Is it ok to leave the code as it is?